src/game/Accounts/AccountMgr.cpp

```diff
@@ -37,6 +38,9 @@ AccountMgr::~AccountMgr()
 
 AccountOpResult AccountMgr::CreateAccount(std::string username, std::string password) const
 {
+    SRP6* srp = new SRP6();
```

Comment:
I see several of these local variables without matching deletes in this commit, so I think you've introduced several memory leaks. Since this is a local variable, just create the variable on the stack instead, it's not such a big object anyway (```SRP6 srp;```). See also remarks below.

Reply:
I'm not sure how `AccountMgr` is instanciated. The reason why I didn't create that variable on the stack was to ensure that sensitive data from SRP computation will not be shared/mixed with other actions. Meanwhile I added the delete statements for `srp`. Are you fine with that?

Reply:
The AccountMgr is a MaNGOS::Singleton, so it's created when the instance is first accessed I think.

As long as we're not leaking memory, I don't mind having it on the heap in cases like this. It probably doesn't add a lot of security (the pointer is on the stack instead, so the data is just one hop away) but it obviously also doesn't make it worse.

Remember that instead of adding delete statements, you can also use std::unique_ptr for example to do that for you (even in the face of exceptions). But I might be barking up a tree for no reason here because I didn't check your updated code yet. I'll try to do that later this week.

Reply:
no man. it's my lack of up2date cpp knowledge. I'll push an updated version after testing